### PR TITLE
Update authSample to use the newest interface

### PR DESCRIPTION
**Pull request by bochunz**
@bsittler @wora PTAL

### CODE REVIEW

File: samples/authSample.html [hunk 1]
```diff
@@ -37,42 +37,45 @@
...
       //   https://console.developers.google.com/apis/credentials?project=_
       // In your API Console project, add a JavaScript origin that corresponds
       //   to the domain where you will be running the script.
       var clientId = 'YOUR_WEB_CLIENT_ID.apps.googleusercontent.com';
 
       // Enter one or more authorization scopes. Refer to the documentation for
-      // the API or https://developers.google.com/identity/protocols/googlescopes
+      // the API or https://developers.google.com/people/v1/how-tos/authorizing
       // for details.
       var scopes = 'profile';
```
Comment (bochunz) [inline samples/authSample.html]: Not sure if we should recommend the short scope "profile" or it's long equivalent "https://www.googleapis.com/auth/userinfo.profile"

Comment (bsittler) [inline samples/authSample.html]: Short (OpenID Connect-style?) version is preferable, I think.

File: samples/authSample.html [hunk 2]
```diff
@@ -37,42 +37,45 @@
...
       }
 
-      function initAuth() {
-        gapi.client.setApiKey(apiKey);
-        gapi.auth2.init({
-            client_id: clientId,
+      function initClient() {
+        gapi.client.init({
+            apiKey: apiKey,
+            discoveryDocs: dircoveryDocs,
```
Comment (bsittler) [inline samples/authSample.html]: typo: dircoveryDocs -> discoveryDocs

File: samples/authSample.html [hunk 3]
```diff
@@ -91,32 +94,28 @@
...
-          });
+        gapi.client.people.people.get({
+          resourceName: 'people/me'
+        }).then(function(resp) {
+          var p = document.createElement('p');
+          var name = resp.result.names[0].givenName;
+          p.appendChild(document.createTextNode('Hello, '+name+'!'));
+          document.getElementById('content').appendChild(p);
         });
-        // Note: In this example, we use the People API to get the current
```
Comment (bsittler) [inline samples/authSample.html]: This comment is still relevant, right? Just move it, I think

File: samples/authSample.html [hunk 4]
```diff
@@ -91,32 +94,28 @@
...
-        // user's name. In a real app, you would likely get basic profile info
-        // from the GoogleUser object to avoid the extra network round trip.
-        console.log(auth2.currentUser.get().getBasicProfile().getGivenName());
       }
     </script>
-    <script src="https://apis.google.com/js/api.js?onload=handleClientLoad"></script>
+    <script async defer src="https://apis.google.com/js/api.js" 
+      onload="handleClientLoad()"
```
Comment (bsittler) [inline samples/authSample.html]: this should also include this.onload = function() {}; to ensure one-shot callback

File: samples/authSample.html (hunk 1 above)
Comment (bochunz) [inline samples/authSample.html]: However if you request the "profile" scope, the actual scope the oauth server gives is the long "https://www.googleapis.com/auth/userinfo.profile" one. Plus I think we stopped to create any short scopes. Maybe we are deprecating the shorts ones?

File: samples/authSample.html (hunk 2 above)
Comment (bochunz) [inline samples/authSample.html]: Done

File: samples/authSample.html (hunk 4 above)
Comment (bochunz) [inline samples/authSample.html]: Added, PTAL

File: samples/authSample.html (hunk 3 above)
Comment (bochunz) [inline samples/authSample.html]: Let remove to let reader focus on API part

File: samples/authSample.html (hunk 1 above)
Comment (iainmcgin) [inline samples/authSample.html]: As profile is explicitly defined in the OpenID spec, I imagine we will continue to support that in perpetuity, but I'll check with the owners of our authorization endpoint.

Comment (iainmcgin) [inline samples/authSample.html]: Confirmed, we'll continue to support the OpenID specified scope strings that map to something meaningful (openid, email, profile). Address and phone are not supported.

**Pull request merged**